pkg/block/blocktest/adapter.go

```diff
@@ -187,18 +188,31 @@ func dumpPathTree(t testing.TB, ctx context.Context, adapter block.Adapter, qk b
 	return tree
 }
 
+func createMultipartFile() ([][]byte, []byte) {
+	const (
+		multipartNumberOfParts = 3
+		multipartPartSize      = 5 * 1024 * 1024
+	)
+	parts := make([][]byte, multipartNumberOfParts)
+	var partsConcat []byte
+	for i := 0; i < multipartNumberOfParts; i++ {
+		parts[i] = randstr.Bytes(multipartPartSize + i)
+		partsConcat = append(partsConcat, parts[i]...)
+	}
+	return parts, partsConcat
+}
```

Comment:
Using standard package
```go
func createMultipartFile() ([][]byte, error) {
	const (
		multipartNumberOfParts = 3
		multipartPartSize      = 5 * 1024 * 1024
	)
	parts := make([][]byte, multipartNumberOfParts)
	for i := 0; i < multipartNumberOfParts; i++ {
		parts[i] = make([]byte, multipartPartSize)
		_, err := rand.Read(parts[i])
		if err != nil {
			return nil, err
		}
	}
	return parts, nil
}
```

Using copy to pass large number of byte (optional outside this function)

```go
	full := make([]byte, len(parts)*len(parts[0]))
	var i int
	for _, p := range parts {
		i += copy(result[i:], p)
	}
```

Reply:
We are already using this package in other places in our code